### PR TITLE
chore: bump Vite and Vitest

### PR DESCRIPTION
**Pull request by juliusmarminge**
Closes #

## 🎯 Changes

What changes are made in this PR? Is it a feature or a bug fix?

## ✅ Checklist

- [ ] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [ ] If necessary, I have added documentation related to the changes made.
- [ ] I have added or updated the tests related to the changes made.

### CODE REVIEW

**Review by KATT — DISMISSED** [review]

File: packages/tests/server/interop/middleware.test.ts [hunk 1]
```diff
@@ -497,11 +497,10 @@ test('omitting ctx in next() does not affect a previous middleware', async () =>
       })
       .query('test', {
         resolve({ ctx }) {
-          expectTypeOf(ctx).toEqualTypeOf<
-            OriginalContext & {
-              user: User;
-            }
-          >();
+          expectTypeOf(ctx).toEqualTypeOf<{
```
Comment (juliusmarminge) [inline packages/tests/server/interop/middleware.test.ts:500]: CleanShot 2023-12-18 at 19 22 42@2x

this is from main, it's the same. I guess the matcher (expect-type) has changed and gotten stricter?

File: packages/tests/server/regression/issue-5034-input-with-index-signature.test.ts [hunk 2]
```diff
@@ -60,6 +60,7 @@ describe('inferRouterInputs/inferRouterOutputs', () => {
     }>();
     expectTypeOf<Output>().toEqualTypeOf<{
       [x: string]: unknown;
+      [x: number]: unknown;
```
Comment (juliusmarminge) [inline packages/tests/server/regression/issue-5034-input-with-index-signature.test.ts:63]: CleanShot 2023-12-18 at 19 22 27@2x

same here

File: packages/server/src/adapters/node-http/content-type/form-data/streamMultipart.js [hunk 3]
```diff
@@ -0,0 +1,502 @@
...
+    try {
+      if (fn1(pos1 + i) !== fn2(pos2 + i)) {
+        return false;
+      }
+    } catch (e) {
+      // no-op
+    }
+  }
+  return true;
+}
```
Comment (juliusmarminge) [inline packages/server/src/adapters/node-http/content-type/form-data/streamMultipart.js:76]: yolo

Comment (juliusmarminge) [reply packages/server/src/adapters/node-http/content-type/form-data/streamMultipart.js:76]: CleanShot 2024-01-07 at 00 38 03@2x
CleanShot 2024-01-07 at 00 39 51@2x

**Review by KATT — APPROVED** [review]

**Pull request merged**